internal/consensus/common_test.go

```diff
@@ -189,34 +190,22 @@ func incrementRound(vss ...*validatorStub) {
 	}
 }
 
-type ValidatorStubsByPower []*validatorStub
-
-func (vss ValidatorStubsByPower) Len() int {
-	return len(vss)
-}
-
-func (vss ValidatorStubsByPower) Less(i, j int) bool {
-	vssi, err := vss[i].GetPubKey(context.TODO())
-	if err != nil {
-		panic(err)
-	}
-	vssj, err := vss[j].GetPubKey(context.TODO())
-	if err != nil {
-		panic(err)
-	}
-
-	if vss[i].VotingPower == vss[j].VotingPower {
-		return bytes.Compare(vssi.Address(), vssj.Address()) == -1
-	}
-	return vss[i].VotingPower > vss[j].VotingPower
-}
+func sortVValidatorStubsByPower(ctx context.Context, vss []*validatorStub) {
+	sort.Slice(vss, func(i, j int) bool {
+		vssi, err := vss[i].GetPubKey(ctx)
+		if err != nil {
+			panic(err)
+		}
+		vssj, err := vss[j].GetPubKey(ctx)
+		if err != nil {
+			panic(err)
+		}
 
-func (vss ValidatorStubsByPower) Swap(i, j int) {
-	it := vss[i]
-	vss[i] = vss[j]
-	vss[i].Index = int32(i)
```

Comment:
I think we may have to preserve this—the default swap is a by-value exchange, and won't update the embedded field.

Reply:
oh god, this cased everything to blow up and I kept looking at the less  function and swap was the thing that did it. :headdesk: